website/blog/2021-08-05-win32component.md

```diff
@@ -0,0 +1,416 @@
+
+In our context, App Service will enable us to create a communication channel between the React Native application and the classic Windows process. The main difference compared to the traditional usage is that the App Service will be hosted and consumed by the same package, since the two processes (UWP and classic Win32) will be packaged together.
+
+App Services support two approaches: [in-proc](https://docs.microsoft.com/windows/uwp/launch-resume/convert-app-service-in-process) (when the implementation of the service is defined in the main application itself) and [out-of-proc](https://docs.microsoft.com/windows/uwp/launch-resume/how-to-create-and-consume-an-app-service) (when the implementation of the service is defined in an external Windows Runtime Component, which gets executed in a different process). For our scenario, we must use the in-proc approach, since we need to hold a direct connection with the main application.
+
+#### A native module
+
+[A native module](https://microsoft.github.io/react-native-windows/docs/native-modules) is the way to expose native features to the JavaScript layer of React Native application. In the Windows context, a native module is a Windows Runtime Component that, through attributes which decorates your code, can expose methods, properties and events to JavaScript.
+The native module is needed as a glue between the React Native world and the UWP world. All the APIs to interact with an App Service are exposed by the Windows Runtime and, as such, we need a middle man that can make them accessible to JavaScript.
+
+#### Windows Application Packaging Project
+
+A [Windows Application Packaging Project](https://docs.microsoft.com//windows/msix/desktop/desktop-to-uwp-packaging-dot-net) (WAP project, from now on) is a template available in Visual Studio, which is typically used to package as MSIX classic Windows application. In our context, we're going to use it to bundle together the classic Windows process and the React Native for Windows application in the same MSIX package.
+
+The diagram below shows how all these components are tight together:
+
+![The architecture of the solution we're going to build in the article](assets/2021-05-08-win32component/architecture.png)
+
+1. The user launches the React Native application which, at the same time, starts also the classic Windows process.
+2. The Windows process, at startup, establishes a communication with the React Native application using an App Service and it stores a reference to the channel.
```

Comment:
Windows => win32 :) here and others

Reply:
I should have fided all the instances :-)